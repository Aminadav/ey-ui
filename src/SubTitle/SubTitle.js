import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import useStyles from './styles';

const SubTitle = ({ children, label }) => {
  const classes = useStyles();
  return (
    <React.Fragment>
      {label && (
        <Typography
          className={classes.label}
          variant="overline"
          gutterBottom
          data-testid="subtitle-label"
        >
          {`${label}: `}
        </Typography>
      )}
      <Typography data-testid="subtitle-content" variant="subtitle1" gutterBottom>
        {children}
      </Typography>
    </React.Fragment>
  );
};

SubTitle.propTypes = {
  label: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.node]).isRequired,
};

SubTitle.defaultProps = {
  label: null,
};

export default SubTitle;
