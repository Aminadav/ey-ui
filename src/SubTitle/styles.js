import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(() => ({
  label: {
    fontWeight: 'bold',
    paddingRight: '1rem',
  },
}));

export default useStyles;
