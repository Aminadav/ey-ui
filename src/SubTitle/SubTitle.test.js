import React from 'react';
import { render } from '@testing-library/react';
import SubTitle from './SubTitle';

describe('SubTitle', () => {
  it('component matches snapshot with a label prop', () => {
    const { container } = render(<SubTitle label="Partner">Partner1</SubTitle>);
    expect(container).toMatchSnapshot();
  });

  it('component matches snapshot without a label prop', () => {
    const { container } = render(<SubTitle>Partner1</SubTitle>);
    expect(container).toMatchSnapshot();
  });

  it('should render properly with a label element', () => {
    const { getByTestId } = render(<SubTitle label="Partner">Partner1</SubTitle>);
    expect(getByTestId('subtitle-label')).toBeInTheDocument();
  });

  it('should render properly with no label element', () => {
    const { queryByTestId } = render(<SubTitle>Partner1</SubTitle>);
    expect(queryByTestId('subtitle-label')).toBeNull();
  });

  it('should render with proper label content', () => {
    const { getByTestId } = render(<SubTitle label="TestLabel">Partner1</SubTitle>);
    expect(getByTestId('subtitle-label')).toHaveTextContent('TestLabel');
  });

  it('should allow for text input as children', () => {
    const { container } = render(<SubTitle>TextInputWithoutDivs</SubTitle>);
    expect(container).toHaveTextContent('TextInputWithoutDivs');
  });

  it('should allow for array input as children', () => {
    const { container } = render(<SubTitle>{['ArrayElement1', 'ArrayElement2']}</SubTitle>);

    expect(container).toHaveTextContent('ArrayElement1');
    expect(container).toHaveTextContent('ArrayElement2');
  });

  it('should allow for node input as children', () => {
    const { getByTestId } = render(
      <SubTitle>
        <div>NodeInputWithDivs</div>
      </SubTitle>,
    );

    expect(getByTestId('subtitle-content')).toContainHTML('NodeInputWithDivs');
  });
});
