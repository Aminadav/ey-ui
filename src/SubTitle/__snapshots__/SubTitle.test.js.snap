// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitle component matches snapshot with a label prop 1`] = `
<div>
  <span
    class="MuiTypography-root makeStyles-label-1 MuiTypography-overline MuiTypography-gutterBottom"
    data-testid="subtitle-label"
  >
    Partner: 
  </span>
  <h6
    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
    data-testid="subtitle-content"
  >
    Partner1
  </h6>
</div>
`;

exports[`SubTitle component matches snapshot without a label prop 1`] = `
<div>
  <h6
    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
    data-testid="subtitle-content"
  >
    Partner1
  </h6>
</div>
`;
